src/metabase/models/dashboard_card.clj

```diff
@@ -35,11 +36,21 @@
                   :visualization_settings {}}]
     (merge defaults dashcard)))
 
+(defn normalize-parameter-mappings
+  "Normalize `parameter-mappings` when coming out of the application database or in via an API request."
+  [parameter-mappings]
+  (or (normalize/normalize-fragment [:parameters] parameter-mappings)
+      []))
+
+(models/add-type! ::parameter-mappings
+  :in  (comp i/json-in normalize-parameter-mappings)
+  :out (comp (i/catch-normalization-exceptions normalize-parameter-mappings) i/json-out-with-keywordization))
+
```

Comment:
One more tangentially related change. I moved the `:parameter-mappings` Toucan column type definition into the DashboardCard namespace since that's the only model that uses it and I needed to use functions powering it elsewhere